src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs

```diff
@@ -151,14 +151,14 @@ public void TestReflectInitializationViaSubscription3()
         [Fact]
         public async Task TestCanceledRequest()
         {
-            using (var eventRecords = new EventObserverAndRecorder())
+            CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
```

Comment:
Previously CancellationTokenSource was cancelled after 1ms. Now it is cancelled when the Start event is received, 10 sec is just a safe belt. 

It makes test deterministic. 
Now ()without this fix) it fails sometimes because on different stages of request processing different exceptions could be thrown. Test also checks that start event is received, which is not guaranteed with 1ms timeout.

The point of the test is to make sure that reflection hook does not do anything unexpected with normal HttpWebRequest processing.